ext/storage_sources/s3_store/test/test_s3_connection.cpp

```diff
@@ -282,34 +282,40 @@ TestGetObject(const Aws::S3Crt::ClientConfiguration &config)
 }
 /*
  * TestObjectExists --
- *     Unit test to check if an object exists in an AWS bucket.
+ *     Unit test to check if an object exists in an AWS bucket and objectSize is correct.
  */
 int
 TestObjectExists(const Aws::S3Crt::ClientConfiguration &config)
 {
     S3Connection conn(config, TestDefaults::bucketName, TestDefaults::objPrefix);
     bool exists = false;
     int ret = TEST_FAILURE;
+    size_t objectSize;
 
     const std::string objectName = "test_object";
     const std::string fileName = "test_object.txt";
 
     /* Create a file to upload to the bucket.*/
     std::ofstream File(fileName);
-    File << "Test payload";
+    std::string payload = "Test payload";
+    File << payload;
     File.close();
 
-    ret = conn.ObjectExists(objectName, exists);
-    if (ret != 0 || exists)
+    ret = conn.ObjectExists(objectName, exists, objectSize);
+    if (ret != 0 || exists || objectSize != 0)
```

Comment:
Would it be better to split this check? I.e.
```
if ((ret = conn.ObjectExists(objectName, exists, objectSize) != 0)
    return (ret);
if (exists || objectSize != 0)
    return (TEST_FAILURE);
```

Reply:
Yep sounds good!